types/react/index.d.ts

```diff
@@ -128,7 +133,7 @@ declare namespace React {
     function createClass<P, S>(spec: ComponentSpec<P, S>): ClassicComponentClass<P>;
 
     function createFactory<P extends DOMAttributes<T>, T extends Element>(
-        type: string): DOMFactory<P, T>;
+        type: keyof ReactHTML): DOMFactory<P, T>;
```

Comment:
Added missing overloads for `createFactory`:
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/17553/commits/929b4a206daefb5463be587e74ea5fa225c40e90#diff-96b72df8b13a8a590e4f160cbc51f40cR136